src/catching_show.c

```diff
@@ -121,19 +119,19 @@ int CatchingShow_GetTypePoints(FieldSystem *fieldSystem)
     return CalculateTypePoints(&sCatchingShow);
 }
 
-static void sub_02056400(u32 species, u8 *param1)
+static void BufferSpeciesData(u32 species, u8 *speciesDataArray)
 {
     GF_ASSERT(0 < species && species <= NATIONAL_DEX_COUNT);
 
-    int v0 = (species - 1) * 6;
+    int speciesOffset = (species - 1) * BYTES_PER_SPECIES;
 
-    NARC_ReadFromMemberByIndexPair(param1, NARC_INDEX_ARC__PPARK, 0, v0, sizeof(u8) * 6);
+    NARC_ReadFromMemberByIndexPair(speciesDataArray, NARC_INDEX_ARC__PPARK, 0, speciesOffset, sizeof(u8) * BYTES_PER_SPECIES);
 }
 
-static void InitEncounterData(FieldSystem *fieldSystem, CatchingShow *catchingShow)
+static void InitSpeciesData(FieldSystem *fieldSystem, CatchingShow *catchingShow)
 {
     int i;
-    u8 v1[8];
+    u8 speciesData[8];
```

Comment:
```suggestion
    PalParkSpecies speciesData;
    u8 unused[2];
```
No idea if this matches.

Reply:
it doesnt look like this matches, which i guess applies to all the recommended changes to introduce this struct.
which is a huge bummer because that suggestion looks much cleaner than the way it is now

Reply:
Might just be some padding on the struct getting added. That is my initial hunch, at least.

Reply:
probably. for kicks i also tried `u32 unused` in the proposed `PalParkSpecies` struct and instead of doing `sizeof(PalParkSpecies)`, hard coded a 6 back in there but still no dice